docs/v4/api.md

```diff
@@ -0,0 +1,43 @@
+# API design document
+
+## Field data input
+
+Here is the proposed API for Field data ingestion into Parcels.
+
+```{mermaid}
+
+classDiagram
+    class Fieldset{
+        +List[Field] fields
+    }
+    class Field{
+        +xr.Dataset|xr.DataArray|ux.Dataset|ux.DataArray data
+        +parcels.Grid|ux.Grid grid
+        +Interpolator interpolator
+        +eval(x,y,z,t,particle)
+    }
+
+    class Interpolator{
+        +assert_is_compatible(Field)
+        +interpolate(Field, rid, x, y, z, t)
```

Comment:
Looks good, and then the barycentric coordinates can be calculated via utility functions if the interpolator author cares about them :)